src/main/kotlin/no/skatteetaten/aurora/boober/feature/ClingerSidecarFeature.kt

```diff
@@ -0,0 +1,191 @@
+import no.skatteetaten.aurora.boober.model.AuroraConfigFieldHandler
+import no.skatteetaten.aurora.boober.model.AuroraContextCommand
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.AuroraResource
+import no.skatteetaten.aurora.boober.model.PortNumbers
+import no.skatteetaten.aurora.boober.service.CantusService
+import no.skatteetaten.aurora.boober.utils.addIfNotNull
+import no.skatteetaten.aurora.boober.utils.boolean
+import no.skatteetaten.aurora.boober.utils.validUrl
+
+val AuroraDeploymentSpec.clingerSidecar: String?
+    get() =
+        this.featureEnabled("azure/proxySidecar") {
+            this["azure/proxySidecar/version"]
+        }
+
+@org.springframework.stereotype.Service
+class ClingerSidecarFeature(
+    cantusService: CantusService
+) : AbstractResolveTagFeature(cantusService) {
+    override fun isActive(spec: AuroraDeploymentSpec): Boolean {
+        val clingerSidecar = spec.clingerSidecar
+
+        return clingerSidecar != null
+    }
+
+    override fun enable(header: AuroraDeploymentSpec): Boolean {
+        return !header.isJob
+    }
+
+    override fun createContext(
+        spec: AuroraDeploymentSpec,
+        cmd: AuroraContextCommand,
+        validationContext: Boolean
+    ): Map<String, Any> {
+
+        val clingerTag = spec.clingerSidecar
+
+        if (validationContext || clingerTag == null) {
+            return emptyMap()
+        }
+
+        return createImageMetadataContext(
+            repo = "no_skatteetaten_aurora",
+            name = "clinger",
+            tag = clingerTag
+        )
+    }
+
+    override fun handlers(header: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraConfigFieldHandler> {
+        return setOf(
+            AuroraConfigFieldHandler(
+                "azure",
+                defaultValue = false,
+                validator = { it.boolean() },
+                canBeSimplifiedConfig = true
+            ),
+            AuroraConfigFieldHandler(
+                "azure/proxySidecar",
+                defaultValue = false,
+                validator = { it.boolean() },
+                canBeSimplifiedConfig = true
+            ),
+            AuroraConfigFieldHandler(
+                "azure/proxySidecar/version",
+                defaultValue = "0.3.1"
+            ),
+            AuroraConfigFieldHandler(
+                "azure/proxySidecar/discoveryUrl",
+                validator = { it.validUrl(required = false) }),
+
+            AuroraConfigFieldHandler(
+                "azure/proxySidecar/ivGroupsRequired",
+                defaultValue = false,
+                validator = { it.boolean() })
+        )
+    }
+
+    override fun modify(
+        adc: AuroraDeploymentSpec,
+        resources: Set<AuroraResource>,
+        context: FeatureContext
+    ) {
+
+        adc.clingerSidecar ?: return
+
+        val container = createClingerProxyContainer(adc, context)
```

Comment:
Jeg tror man trenger en mekanisme for å skru dette av i ocp3. AuroraConfig må validere på OCP3, men deploy må feile.

Reply:
Dersom du får en clinger-sidecar i OCP3, så vil den ikke virke (om ikke annet så pga manglende LDAP-passord), og applikasjonen vil ikke få trafikk.  Så om man deployer en slik applikasjon finner man rimelig fort ut at det ikke går. Tatt i betraktning at vi "snart" er over på OCP4, trenger vi å feile eksplisitt ved deploy-tid?